src/Mutator/MutatorFactory.php

```diff
@@ -82,16 +82,15 @@ public function create(array $resolvedMutators): array
                     self::getConfigurableMutator($mutatorClassName, $settings) :
                     new $mutatorClassName();
 
-            if ($ignored === []) {
-                $mutators[$mutator->getName()] = $mutator;
+            if ($ignored !== []) {
+                $mutator = new IgnoreMutator(new IgnoreConfig($ignored), $mutator);
+            }
```

Comment:
It appears to me the mutator factory does slightly more than it should. There could be a thin decorator object, that would be super easy to test.

In the same light adding `IgnoreMutator` could a job of a similar object, yet we're already here so it might be easier to leave it as it is for time being.

Reply:
TBH, I didn't get the idea here, or this file still looks ok for me. I left it as is for now.